src/prefect/blocks/core.py

```diff
@@ -1086,34 +1104,45 @@ async def register_blocks_in_annotation(annotation: type) -> None:
                     await register_blocks_in_annotation(inner_annotation)
 
         for field in cls.model_fields.values():
-            await register_blocks_in_annotation(field.annotation)
+            if field.annotation:
+                await register_blocks_in_annotation(field.annotation)
+
+        local_block_type = cls._to_block_type()
 
         try:
-            block_type = await client.read_block_type_by_slug(
+            block_type = await _client.read_block_type_by_slug(
                 slug=cls.get_block_type_slug()
             )
 
             cls._block_type_id = block_type.id
 
-            local_block_type = cls._to_block_type()
             if _should_update_block_type(
                 local_block_type=local_block_type, server_block_type=block_type
             ):
-                await client.update_block_type(
-                    block_type_id=block_type.id, block_type=local_block_type
+                await _client.update_block_type(
+                    block_type_id=block_type.id,
+                    block_type=BlockTypeUpdate.model_construct(
+                        **local_block_type.model_dump()
+                    ),
```

Comment:
It might be worthwhile to add a `to_block_type_update` to `BlockType` to avoid a `model_dump`.